dask/array/tests/test_cupy.py

```diff
@@ -931,6 +932,303 @@ def test_bincount():
+def test_vindex():
+    x_np = np.arange(56).reshape((7, 8))
+    x_cp = cupy.arange(56).reshape((7, 8))
+
+    d_np = da.from_array(x_np, chunks=(3, 4))
+    d_cp = da.from_array(x_cp, chunks=(3, 4))
+
+    res_np = da.core._vindex(d_np, [0, 1, 6, 0], [0, 1, 0, 7])
+    res_cp = da.core._vindex(d_cp, [0, 1, 6, 0], [0, 1, 0, 7])
+
+    assert type(res_cp._meta) == cupy.core.core.ndarray
+    assert_eq(res_cp, res_cp)  # Check that _meta and computed arrays match types
+
+    assert_eq(res_np, res_cp)
+
+
+@pytest.mark.skipif(np.__version__ < "1.20", reason="NEP-35 is not available")
+def test_percentile():
+    d = da.from_array(cupy.ones((16,)), chunks=(4,))
+    qs = np.array([0, 50, 100])
+
+    assert_eq(
+        da.percentile(d, qs, interpolation="midpoint"),
+        np.array([1, 1, 1], dtype=d.dtype),
+    )
+
+    x = cupy.array([0, 0, 5, 5, 5, 5, 20, 20])
+    d = da.from_array(x, chunks=(3,))
+
+    result = da.percentile(d, qs, interpolation="midpoint")
+    assert_eq(result, np.array([0, 5, 20], dtype=result.dtype))
+
+    # Currently fails, tokenize(cupy.array(...)) is not deterministic.
+    # See https://github.com/dask/dask/issues/6718
+    # assert same_keys(
+    #     da.percentile(d, qs),
+    #     da.percentile(d, qs)
+    # )
+
+    assert not same_keys(
+        da.percentile(d, qs, interpolation="midpoint"),
+        da.percentile(d, [0, 50], interpolation="midpoint"),
+    )
+
+
+@pytest.mark.skipif(np.__version__ < "1.20", reason="NEP-35 is not available")
+def test_percentiles_with_empty_arrays():
+    x = da.from_array(cupy.ones(10), chunks=((5, 0, 5),))
+    res = da.percentile(x, [10, 50, 90], interpolation="midpoint")
+
+    assert type(res._meta) == cupy.core.core.ndarray
+    assert_eq(res, res)  # Check that _meta and computed arrays match types
+    assert_eq(res, np.array([1, 1, 1], dtype=x.dtype))
+
+
+@pytest.mark.skipif(np.__version__ < "1.20", reason="NEP-35 is not available")
+def test_percentiles_with_empty_q():
+    x = da.from_array(cupy.ones(10), chunks=((5, 0, 5),))
+    result = da.percentile(x, [], interpolation="midpoint")
+
+    assert type(result._meta) == cupy.core.core.ndarray
+    assert_eq(result, result)  # Check that _meta and computed arrays match types
+    assert_eq(result, np.array([], dtype=x.dtype))
+
+
+@pytest.mark.skipif(np.__version__ < "1.20", reason="NEP-35 is not available")
+@pytest.mark.parametrize("q", [5, 5.0, np.int64(5), np.float64(5)])
+def test_percentiles_with_scaler_percentile(q):
+    # Regression test to ensure da.percentile works with scalar percentiles
+    # See #3020
+    d = da.from_array(cupy.ones((16,)), chunks=(4,))
+    result = da.percentile(d, q, interpolation="midpoint")
+
+    assert type(result._meta) == cupy.core.core.ndarray
+    assert_eq(result, result)  # Check that _meta and computed arrays match types
+    assert_eq(result, np.array([1], dtype=d.dtype))
+
+
+@pytest.mark.skipif(np.__version__ < "1.20", reason="NEP-35 is not available")
```

Comment:
```suggestion
@pytest.mark.skipif(np.__version__ < "1.20", reason="NEP-35 is not available")
```

I would move this into a global variable, perhaps in a settings-like module. That way it's easy to change the test.

Reply:
Specifically, create a variable named `NEP35_AVAILABLE`.

Reply:
Good point, this exists already in https://github.com/dask/dask/blob/ac2e582198ee19a1de81a3c26d950b766b42de85/dask/array/numpy_compat.py#L11 which should suffice.

Reply:
Done in https://github.com/dask/dask/pull/6738/commits/bae0dd67e338152d55e0ea4823a0661c804c3cea

Reply:
I would suggest testing the protocol rather than the version. Something like:

```
def _test_nep35():
    class MyClass:
        def __array_function__(self, func, args, kwargs):
            return True
    try:
        return np.asarray([], like=MyClass())
    except:
        return False
```

Reply:
I would personally prefer they fail to force us to update the tests should anything change, rather than having tests getting skipped. Happy to change my mindset should there be compelling arguments in favor of skipping. 🙂